src/core/lib/transport/metadata_batch.h

```diff
@@ -305,6 +305,13 @@ struct GrpcTagsBinMetadata : public SimpleSliceBasedMetadata {
   static absl::string_view key() { return "grpc-tags-bin"; }
 };
 
+// XEnvoyPeerMetadata
+struct XEnvoyPeerMetadata : public SimpleSliceBasedMetadata {
```

Comment:
How frequently does this change? Is it expected to be constant over a single client<->server pair? Changing between a few different values?

Reply:
(also how big is this expected to be... min/median/max bytes? - a rough guess would suffice)

Reply:
If there are no proxies in the middle, it will always be the same for each client/server pair.  The problem is that we can't depend on that, because if there is a proxy in between, then the proxy could route each request from a given client to a different server (i.e., the client might see a different value for each request), and the proxy could send requests from different clients on the same connection to a server (i.e., the server might see a different value for each request).

FWIW, keep in mind that in the long run, we want to completely eliminate this mechanism in favor of some out-of-band lookup for the metadata, so hopefully this isn't going to stick around forever.  But there are obviously no guarantees.

Reply:
The metadata that we send remains the same for the lifetime of the binary.  The metadata that we receive as Mark mentioned is potentially changing.

Reply:
Ok... and the other question: how big is it?
(I'm trying to sanity check whether the compression policy makes sense)

Reply:
So, the format of the payload is a base64 encoded serialized proto struct.. For my test values, the size was 300 bytes.